app/code/Magento/Quote/Model/CustomerManagement.php

```diff
@@ -100,11 +105,47 @@ public function populateCustomerInfo(QuoteEntity $quote)
         }
     }
 
+    /**
+     * Validate 'CustomerAddressId' in quote for a newly created customer.
+     *
+     * @param QuoteEntity $quote
+     * @return void
+     */
+    private function validateCustomerAddressId(QuoteEntity $quote): void
```

Comment:
I wrote it above - but probably you missed it - this method isn't doing any validation, it's just filling it